Tests/ExponentialBackoffRetryPolicyTests.swift

```diff
@@ -0,0 +1,320 @@
+//
+//  CacheTests.swift
+//
+//  Copyright (c) 2014-2018 Alamofire Software Foundation (http://alamofire.org/)
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+@testable import Alamofire
+import Foundation
+import XCTest
+
+class ExponentialBackoffRetryPolicyTestCase: BaseTestCase {
+
+    // MARK: - Helper Types
+
+    private class StubRequest: DataRequest {
+        let urlRequest: URLRequest
+        override var request: URLRequest? { return urlRequest }
+
+        let mockedResponse: HTTPURLResponse?
+        override var response: HTTPURLResponse? { return mockedResponse }
+
+        init(_ url: URL, method: HTTPMethod, response: HTTPURLResponse?, session: Session) {
+            mockedResponse = response
+
+            let request = Session.RequestConvertible(
+                url: url,
+                method: method,
+                parameters: nil,
+                encoding: URLEncoding.default,
+                headers: nil
+            )
+
+            urlRequest = try! request.asURLRequest()
+
+            super.init(
+                convertible: request,
+                underlyingQueue: session.rootQueue,
+                serializationQueue: session.serializationQueue,
+                eventMonitor: session.eventMonitor,
+                interceptor: nil,
+                retryPolicies: [],
+                delegate: session
+            )
+        }
+    }
```

Comment:
Pretty slick how this turned out. Rather than having to try and pump an entire request/response through the AF system, I just created a stub that overrides the `request` and `response` properties.